packages/app/src/cli/services/dev/migrate-app-module.test.ts

```diff
@@ -17,30 +16,55 @@ function getLocalExtension(attributes: Partial<LocalSource> = {}) {
 function getRemoteExtension(attributes: Partial<RemoteSource> = {}) {
   return {
     uuid: '1234',
-    type: 'payments_app_credit_card',
     title: 'a-different-extension',
     ...attributes,
   } as unknown as RemoteSource
 }
+const defaultMap = {
+  payments_extension: [
+    'payments_app',
+    'payments_app_credit_card',
+    'payments_app_custom_credit_card',
+    'payments_app_custom_onsite',
+    'payments_app_redeemable',
+  ],
+  marketing_activity: ['marketing_activity_extension'],
+}
+
+const defaultIdentifiers = {
+  'module-A': '1234',
+  'module-B': '',
+}
 
-describe('getPaymentsExtensionsToMigrate()', () => {
-  const defaultIds = {
-    'my-action': '1234',
-  }
+describe('getModulesToMigrate()', () => {
+  test('returns an empty array if no extensions are provided', () => {
+    const toMigrate = getModulesToMigrate([], [], {}, defaultMap)
+    expect(toMigrate).toStrictEqual([])
+  })
 
-  test('matching my remote title and localIdentifier', () => {
+  test('matching by remote title and localIdentifier, without defaultIdentifiers', () => {
     // Given
-    const localExtension = getLocalExtension({type: 'payments_extension', localIdentifier: 'my-action'})
-    const remoteExtension = getRemoteExtension({type: 'payments_app_credit_card', title: 'my-action', uuid: 'yy'})
+    const localExtension = getLocalExtension({type: 'payments_extension', localIdentifier: 'module-A'})
+    const localExtensionB = getLocalExtension({type: 'marketing_activity', localIdentifier: 'module-B'})
+    const remoteExtension = getRemoteExtension({type: 'payments_app_credit_card', title: 'module-A', uuid: 'yy'})
+    const remoteExtensionB = getRemoteExtension({type: 'marketing_activity_extension', title: 'module-B', uuid: 'xx'})
 
     // When
-    const toMigrate = getPaymentsExtensionsToMigrate([localExtension], [remoteExtension], defaultIds)
+    const toMigrate = getModulesToMigrate(
+      [localExtension, localExtensionB],
+      [remoteExtension, remoteExtensionB],
+      {},
+      defaultMap,
+    )
 
     // Then
-    expect(toMigrate).toStrictEqual([{local: localExtension, remote: remoteExtension}])
+    expect(toMigrate).toStrictEqual([
+      {local: localExtension, remote: remoteExtension},
+      {local: localExtensionB, remote: remoteExtensionB},
+    ])
   })
 
-  test('matching my remote title and localIdentifier by truncating the title', () => {
+  test('matching my remote title by truncating the title', () => {
```

Comment:
What do we think about changing the extension types to use other ones we've added here, such as flow?